device/modbus_mapper/README.md

```diff
@@ -0,0 +1,147 @@
+# Modbus Mapper
+
+
+## Introduction
+
+Mapper is an application that is used to connect and control devices. This is an implementation of mapper for 
+Modbus protocol. The aim is to create an application through which users can easily operate devices using ModbusTCP/ModbusRTU protocol for communication to the KubeEdge platform. The user is required to provide the mapper with the information required to control their device through the dpl configuration file. These can be changed at runtime by updating configmap.
+
+## Running the mapper
+
+  1. Please ensure that Modbus device is connected to your edge node
+  2. Set 'modbus=true' label for the node (This label is a prerequisite for the scheduler to schedule modbus_mapper pod on the node)
+
+      ```shell
+      kubectl label nodes <name-of-node> modbus=true
+      ```
+
+  3. Build and deploy the mapper by following the steps given below.
+
+### Building the modbus mapper
+
+ ```shell
+cd $GOPATH/src/github.com/kubeedge/kubeedge/device/modbus_mapper
+make # or `make modbus_mapper`
+docker tag modbus_mapper:v1.0 <your_dockerhub_username>/modbus_mapper:v1.0
+docker push <your_dockerhub_username>/modbus_mapper:v1.0
+
+Note: Before trying to push the docker image to the remote repository please ensure that you have signed into docker from your node, if not please type the followig command to sign in
+ docker login
+ # Please enter your username and password when prompted
+```
+
+### Deploying modbus mapper application
+
+```shell
+cd $GOPATH/src/github.com/kubeedge/kubeedge/device/modbus_mapper
+
+# Please enter the following details in the modbus-dpl-configmap.yaml :-
+#    1. Replace <edge_node_name> with the name of your edge node at meta.name
+#    2. Replace <dpl-configuration> with your device protocol configuration at data.dpl
+
+kubectl create -f modbus-dpl-configmap.yaml
+
+# Please enter the following details in the deployment.yaml :-
+#    1. Replace <edge_node_name> with the name of your edge node at spec.template.spec.voluems.configMap.name
+#    2. Replace <your_dockerhub_username> with your dockerhub username at spec.template.spec.containers.image
+
+kubectl create -f deployment.yaml
+```
+
+## Modules
+
+The modbus mapper consists of the following four major modules :-
+
+ 1. Controller
```

Comment:
We have the mapper design : https://github.com/kubeedge/kubeedge/blob/master/docs/proposals/mapper-design.md

And a bluetooth protocol mappers implemented based on this design : 

https://github.com/kubeedge/kubeedge/tree/master/device/bluetooth_mapper

Is it possible that we can model the modbus mappers based on the mapper design architecture ?

Reply:
Modbus mapper is developed by nodejs. It is a little hard to model in that way. I checked with Kevin it's ok if it works with device-crd. Maybe reconstruction by golang later can model according to the above architecture.

Reply:
ok